src/medCore/views/medAbstractLayeredView.cpp

```diff
@@ -463,16 +463,20 @@ void medAbstractLayeredView::write(QString& path)
 
         //getting a working file extension
         //1. find suitable writers
+        QList<QString> allWriters = medAbstractDataFactory::instance()->writers();
         QHash<QString, dtkAbstractDataWriter*> possibleWriters=medDataManager::instance()->getPossibleWriters(layerData(i));
 
-
         //2. use these writers to get a suitable file extension
         QString fileExtension;
-        foreach(QString key,possibleWriters.keys())
-        {
-            if(!possibleWriters[key]->supportedFileExtensions().isEmpty())
+        // we use allWriters as the list of keys to make sure we traverse possibleWriters
+        // in the order specified by the writers priorities.
+        foreach(QString type, allWriters) {
+            if (!possibleWriters.contains(type))
+                continue;
```

Comment:
(sorry if I didn't reply, github is still selectively sending me notifications)
Just to explain why: ```continue``` is a form of ```goto``` statement, and it is widely accepted that ```goto``` statements should be avoided if there are simple alternatives. (It's not always easily avoidable, like using ```break``` in ```switch``` statements).